### ISSUE TITLE
access Webex site on client side

### ISSUE THREAD

**Issue by bbodenmiller** [mattermost/mattermost-plugin-webex#22]
Many Mattermost servers are not allowed to access the outside internet by default. This plugin should support this and access Webex site from client side.

**Comment by cpoile** [comment]
@bbodenmiller This turns out to be how connections to webex work currently. The plugin generates a link and posts it in a message, and when a user clicks on the link in a message they connect directly to the webex system -- there's no traffic through the Mattermost instance during/after the link click.

Please let me know if I'm misunderstanding the issue?

**Comment by bbodenmiller** [comment]
I get that but is the sever or client connecting to Webex to generate the
link in the first place? My sever isn't able to access any content on
*.Webex.com.

**Comment by levb** [comment]
@bbodenmiller ACK. Historically, we focused on server-side and/or server-centric plugins; largely to enable functionality on mobile.

However, there's nothing really, at least in theory, preventing us from creating 100%-webapp plugins (well, serving them from the server). We should consider.

**Comment by bbodenmiller** [comment]
@levb right, I've found the Mattermost move towards more sever side connections when client side would be possible problematic. In security sensitive environments server connections to outside internet are more restricted (blocked entirely in many cases) than client side connections to outside internet.

**Comment by cpoile** [comment]
@levb We could have a back-up webex-client code on the front end. It would query the webex server for the room url given the meeting starter's preferred setting. I think it can be a HW.

**Comment by levb** [comment]
It is my understanding that the client was already not restricted and has no issue connecting to WebEx; the server (plugin!) must use a proxy though. Client can be configured separately anyway.

**Comment by cpoile** [comment]
You're right, the client can do the connection, but the client will need to respect the meeting starter's preferred way to find their meeting room URL (roomId, username/webexId, or email). Then the client can query the Webex server to get that URL.
